src/domain/bulkTransfer/index.js

```diff
@@ -50,10 +56,91 @@ const checkDuplicate = async (bulkTransferId, hash, bulkTransferFulfilmentId = f
   }
 }
 
+const getBulkTransferById = async (id) => {
+  try {
+    let bulkTransfer = await BulkTransferModel.getById(id)
+    let bulkTransferExtensions = await BulkTransferExtensionModel.getByBulkTransferId(id)
+    let individualTransfers = await IndividualTransferModel.getAllById(id)
+    let payeeIndividualTransfers = []
+    individualTransfers = await Promise.all(individualTransfers.map(async (transfer) => {
+      return new Promise(async (resolve, reject) => {
+        let extensions = await IndividualTransferExtensionModel.getByTransferId(transfer.transferId)
+        let extension
+        let result = {
+          transferId: transfer.transferId
+        }
+        if (transfer.fulfilment) {
+          result.fulfilment = transfer.fulfilment
+        }
+        if (transfer.errorCode) {
+          result.errorInformation = {
+            errorCode: transfer.errorCode,
+            errorDescription: transfer.errorDescription
+          }
+        }
+        if (extensions.length > 0) {
+          if (!transfer.fulfilment) {
+            extension = extensions.map(ext => {
+              return { key: ext.key, value: ext.value }
+            })
+          } else {
+            extension = extensions.filter(ext => {
+              return !!ext.transferFulfilmentId
+            }).map(ext => {
+              return { key: ext.key, value: ext.value }
+            })
+          }
+        }
+        if (extension.length > 0) {
+          result.extensionList = { extension }
+        }
+        const allowedPayeeTransfers = [
+          Enum.TransferStateEnum.RESERVED,
+          Enum.TransferStateEnum.COMMITTED
+        ]
+        if (allowedPayeeTransfers.indexOf(transfer.transferStateEnum) !== -1) {
+          payeeIndividualTransfers.push(result)
+        }
+        return resolve(result)
+      })
+    }))
+    let bulkResponse = {
+      bulkTransferId: bulkTransfer.bulkTransferId,
+      bulkTransferState: bulkTransfer.bulkTransferStateId
+    }
+    if (bulkTransfer.completedTimestamp) {
+      bulkResponse.completedTimestamp = bulkTransfer.completedTimestamp
+    }
+    let payerBulkTransfer = Object.assign({}, { destination: bulkTransfer.payerFsp }, bulkResponse)
```

Comment:
I think this can be simplified with:
```js
 let payerBulkTransfer = {destination: bulkTransfer.payerFsp, ...bulkResponse}
```

Reply:
Thanks for reminding me about the Spread syntax, @lewisdaly... Probably I should also revisit other Object.assigns as I continue implementation.